### PR TITLE
Defer checking of class static blocks

### PR DESCRIPTION
**Pull request by Andarist**
closes https://github.com/microsoft/TypeScript/issues/52892

This only helps with a specific situation when a type alias defined in a class static block triggered the first variance check in those circular situations. Then all properties referencing `A` again were reentering `getVariancesWorker` and thus it was assumed that they were OK. My comment at the bottom of this post explains more about this kind of situation: https://github.com/microsoft/TypeScript/issues/59606#issue-2461841606

In other words, this only mitigates the problem in this specific situation but it's still possible to run into problems in similar circular situations.

### CODE REVIEW

File: tests/cases/fourslash/quickInfoAvoidsClassVarianceCircularity1.ts
Comment (Andarist) [inline tests/cases/fourslash/quickInfoAvoidsClassVarianceCircularity1.ts:1]: this one was already fixed by https://github.com/microsoft/TypeScript/pull/53549 , this is just an extra test case

**Comment by RyanCavanaugh** [comment]
@typescript-bot test it

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -47937,7 +47937,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.MethodSignature:
                 return checkMethodDeclaration(node as MethodDeclaration | MethodSignature);
             case SyntaxKind.ClassStaticBlockDeclaration:
-                return checkClassStaticBlockDeclaration(node as ClassStaticBlockDeclaration);
+                return checkNodeDeferred(node);
```
Comment (jakebailey) [inline src/compiler/checker.ts]: Maybe a stupid question, but why is it that this matters here, yet MethodSignatures are not deferred?

Comment (jakebailey) [reply src/compiler/checker.ts]: As in, can't we write that same local type declaration in a method and it breaks? Or is that fine because the _body_ gets deferred?

Comment (Andarist) [reply src/compiler/checker.ts]: My understanding is that:
1. The same local type declaration in a method shouldn't suppress the error either.
2. Object literal methods are deferred but class methods are not. If I try to defer class methods... a ton of tests break. So there has to be a reason behind why they are not deferred. Perhaps it's important for the variance worker?
3. Class static blocks conceptually execute after the class declaration and they shouldn't affect the variance measurement and shouldn't trigger circularities in a way class methods/properties can. So it should be OK to defer them. They should behave in an almost same way as statements executed after the class. They only get a special treatment when it comes to class type parameters and `this`. But it feels like it shouldn't break anything if those get simply resolved in a deferred manner - after the class declaration gets processed.
4. Right now it looks like this PR doesn't do anything because a bug has been introduced in the meantime: https://github.com/microsoft/TypeScript/issues/61982
